media/client/ipc/source/IpcClient.cpp

```diff
@@ -159,11 +159,17 @@ void IpcClient::processIpcThread()
     {
         RIALTO_CLIENT_LOG_ERROR("The ipc channel unexpectedly disconnected, destroying the channel");
 
+        // The test case IpcClientTest.UnexpectedDisconnectWithNotification
```

Comment:
This seems like a test issue rather than a source code issue. Can we improve the test case to fix this instead? We would probably need to wait in UnexpectedDisconnectWithNotification for onConnectionBroken to be executed before completing the test.

Reply:
OPTION A
=======

The test is to determine if onConnectionBroken() is called. If the test fails (and the onConnectionBroken is not called) then the test case could wait forever. But perhaps this is OK because it would cause someone to look into the issue and fix it.

OPTION B
=======

Alternatively we could wait a pre-determined amount of time (10 microseconds or some other magic number) for onConnectionBroken to be called. This would be achieved by simply changing the test case like this...

`--- a/tests/media/client/ipc/ipcClient/IpcClientTest.cpp
+++ b/tests/media/client/ipc/ipcClient/IpcClientTest.cpp
@@ -131,6 +131,7 @@ TEST_F(IpcClientTest, UnexpectedDisconnectWithNotification)
     while (m_channelMock.use_count() == ipcChannelCount)
     {
     }
+    usleep(10); // Delay the destruction of the onConnectionBroken() listener

     // On destruction IpcClient does not disconnect
 } `

======

Please let me know which of these options, A or B, seems best

Reply:
I prefer option A as its deterministic, I would had a timeout to the wait, say 5sec, and fail the test on timeout